salt/modules/win_pkg.py

```diff
@@ -637,16 +637,37 @@ def _get_repo_details(saltenv):
 
         # Do some safety checks on the repo_path as its contents can be removed,
         # this includes check for bad coding
-        paths = (
-            r'[a-z]\:\\$',
-            r'\\$',
-            re.escape(os.environ.get('SystemRoot', r'C:\Windows'))
+        system_root = os.environ.get('SystemRoot', r'C:\Windows')
+        deny_paths = (
+            r'[a-z]\:\\$',  # C:\, D:\, etc
+            r'\\$',  # \
+            re.escape(system_root)  # C:\Windows
         )
-        for path in paths:
-            if re.match(path, local_dest, flags=re.IGNORECASE) is not None:
-                raise CommandExecutionError(
-                    'Local cache dir {0} is not a good location'.format(local_dest)
-                )
+
+        # Since the above checks anything in C:\Windows, there are some
+        # directories we may want to make exceptions for
+        allow_paths = (
+            re.escape('\\'.join([system_root, 'TEMP'])),  # C:\Windows\TEMP
+        )
+
+        # Check the local_dest to make sure it's not one of the bad paths
+        good_path = True
+        for d_path in deny_paths:
+            if re.match(d_path, local_dest, flags=re.IGNORECASE) is not None:
+                # Found deny path
+                good_path = False
+
+        # If local_dest is one of the bad paths, check for exceptions
+        if not good_path:
+            for a_path in allow_paths:
+                if re.match(a_path, local_dest, flags=re.IGNORECASE) is not None:
+                    # Found exception
+                    good_path = True
+
+        if not good_path:
+            raise CommandExecutionError(
+                'Local cache dir {0} is not a good location'.format(local_dest)
```

Comment:
I know this exception pre-dates this PR, but is there a way we can improve it? It's not clear what this exception is actually telling the user, and what they would do to remedy it.